engine/robot.py

```diff
@@ -330,10 +361,53 @@ def set_phase(self, new_phase):
         with open(CSV_PATH + '/phases.csv', 'a') as fd:
             fd.write(str(self.phase) + '\n')
 
-    def execute_avoid_obstacle(self):
-        # TODO: SET BACK TO ORIGINAL MISSION (TRAVERSE OR RETURN)
+    def track_obstacle(self):
         pass
 
+    def execute_avoid_obstacle(self, dist_to_goal, prev_phase):
+        pass
+
+    def execute_boundary_following(self, min_dist):
+        front_dist = self.rf_ultrasonic.distance()
+        back_dist = self.rb_ultrasonic.distance()
+        margin = 1  # Will change
+        forward_dist = 0.01  # Will change
+        turn_angle = math.pi / 90  # Will change
+
+        if math.abs(front_dist - back_dist) < margin:
```

Comment:
I see that there's one value for turn_angle. Will there be support for turning the other way?